publication_monitor.cpp

```diff
@@ -142,6 +151,21 @@ void PublicationMonitor::on_data_available(DDS::DataReader_ptr reader)
 
 } // End PublicationMonitor::on_data_available
 
+bool PublicationMonitor::get_dynamic_type(DDS::DynamicType_var& type, const DDS::BuiltinTopicKey_t& key,
+                                          const char* topic_name, const char* type_name)
+{
+    OpenDDS::DCPS::DomainParticipantImpl* participant =
```

Comment:
Not a huge deal, probably, but I would capture the DomainParticipant_var here so the reference count doesn't get decremented until the function exits. Then you can cast var.in() to a DomainParticipantImpl* as a separate variable if you need that for `get_dynamic_type()`.